src/wallet/wallet2.cpp

```diff
@@ -3125,8 +3128,20 @@ void wallet2::process_parsed_blocks(uint64_t start_height, const std::vector<cry
     num_txes += 1 + parsed_blocks[i].txes.size();
   tx_cache_data.resize(num_txes);
   size_t txidx = 0;
+  crypto::hash prev_block_id;
+  if (start_height > 0) {
+    prev_block_id = m_blockchain[start_height - 1];
```

Comment:
There is a check above that `start_height` is in bounds, but you never check that `start_height - 1` is in bounds. It must be checked too.

Reply:
I guess you just shouldn't init `prev_block_id` if it's not in bounds, instead of throwing an exception like above.

Reply:
You're right. It could fail if the blockchain is trimmed to one block. I just fixed it.